### PR TITLE
Input schema validation support for Pydantic and Marshmellow 

### PR DESCRIPTION
**Pull request by tabotkevin**
check `examples/schema_validation.py`

"""
Decorator for parsing and validating input schema.
   Supports both Pydantic and Marshmallow.

Usage::
    import time

    from pydantic import BaseModel
    import responder

    class Item(BaseModel)
        name: str

    api = responder.API()

    @api.route("/create")
    @api.trust(Item)
    def create_item(req, resp, *, data):
        @api.background.task
        def process_item(item):
            time.sleep(2)
            print(item)   # e.g {"name": "Monster Hunter"}

        process_item(data)
        resp.media = {"msg": "created"}
"""

## Summary by CodeRabbit

- **New Features**
  - Introduced schema validation for creating books with support for Pydantic and Marshmallow libraries.
  - Implemented a new `trust` decorator for input schema parsing and validation in the API.

- **Enhancements**
  - Updated API metadata including title, version, and documentation details.
  - Improved the `Response` class header assignment for better readability.

- **Documentation**
  - Added testing instructions within the example Python web service code.

- **Tests**
  - Added new test cases for Pydantic and Marshmallow schema validation.

- **Dependencies**
  - Included `pydantic` in the project's dependencies for data validation.
  - Added `pytest` and `pytest-mock` to the development dependencies for enhanced testing capabilities.

- **Style**
  - Standardized string quotations and improved code formatting in example files.

### CODE REVIEW

File: responder/models.py [hunk 1]
```diff
@@ -335,7 +335,7 @@ async def body(self):
 
         for format in self.formats:
             if self.req.accepts(format):
-                data = (await self.formats[format](self, encode=True)), {}
+                data = (await self.formats[format](self, encode=True)), {}  # noqa: F841
```
Comment (kennethreitz) [inline responder/models.py:338]: Let's remove the noqa.

File: responder/ext/schema/__init__.py [hunk 2]
```diff
@@ -81,7 +81,10 @@ def _apispec(self):
                 spec.path(path=route.route, operations=operations)
 
         for name, schema in self.schemas.items():
-            spec.components.schema(name, schema=schema)
+            if hasattr(schema, "schema"):
+                spec.components.schema(name, schema.schema())  # Pydantic
```
Comment (kennethreitz) [inline responder/ext/schema/__init__.py:85]: comment above.

Comment (kennethreitz) [reply responder/ext/schema/__init__.py:85]: or just add a period after actually. and let's have them be lower cased.

Comment (kennethreitz) [reply responder/ext/schema/__init__.py:85]: so like `code   # pydantic.

**Comment by kennethreitz** [comment]
This is beautiful.

**Comment by kennethreitz** [comment]
let's make pydantic a dependency

**Comment by tabotkevin** [comment]
Yes, I added `pydantic` in the setup.py file for installation.

File: examples/schema_validation.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+
+# Let's make an HTTP request to the server, to test it out.'}
+r = api.requests.post("http://;/pybook", json={"price": 9.99, "title": "Pydantic book"})
+print(r.text)
+
+r = api.requests.post(
+    "http://;/marshbook", json={"price": 10.99, "title": "Marshmallow book"}
+)
+print(r.text)
```
[the remark this answers is not in the document]
Comment (kennethreitz) [reply examples/schema_validation.py:80]: no, we must use ';'

**Pull request merged**